lib/opal/cli_runners/chrome.rb

```diff
@@ -53,6 +59,38 @@ def run
 
       private
 
+      def prepare_files_in(dir)
+        # Chrome can't handle huge data passed to `addScriptToEvaluateOnLoad`
+        # https://groups.google.com/a/chromium.org/forum/#!topic/chromium-discuss/U5qyeX_ydBo
+        # The only way is to create temporary files and pass them to chrome.
+        File.write("#{dir}/source-maps.js", File.read("#{__dir__}/source-map-support-browser.js"))
+        File.write("#{dir}/index.js", builder.to_s)
+        File.write("#{dir}/index.html", <<~HTML)
+          <html>
+          <head>
+            <meta charset='utf-8'>
+            <script src='./source-maps.js'></script>
+            <script>
+            sourceMapSupport.install({
+              retrieveSourceMap: function(path) {
+                if (path.endsWith('/index.js')) {
+                  return {
+                    url: './index.map',
+                    map: #{builder.source_map.to_json.to_json}
```

Comment:
can the second `to_json` be replaced with `inspect`?

Reply:
I'm not 100% sure the escaping is the same between the two… I since moved the generation of the values to be written to the top of the method, so the two `to_json` don't appear one after the other anymore.